app/notification.go

```diff
@@ -191,6 +191,17 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 			go func(userID string) {
 				defer close(mac)
 				_, incrementMentions := mentions.Mentions[userID]
+				// if the user was not explicitly mentioned, check if they explicitly unfollowed the thread
+				if !incrementMentions {
+					membership, err := a.Srv().Store.Thread().GetMembershipForUser(userID, post.RootId)
+					if nfErr := new(store.ErrNotFound); err != nil && !errors.As(err, &nfErr) {
```

Comment:
This like is very confusing, I think it shouldn't be inlined everything.

Reply:
@jespino this is cleaner?
```
					var nfErr *store.ErrNotFound
		
					if err != nil && !errors.As(err, &nfErr) {
						mac <- model.NewAppError("SendNotifications", "app.channel.autofollow.app_error", nil, err.Error(), http.StatusInternalServerError)
						return
					}
```

Reply:
Yes it is.